### DISCUSSION TITLE
404 page is not localized

### DISCUSSION THREAD

**Discussion by DmitryRyumin** [george-gca/multi-language-al-folio#28] [Q&A] [answered]
Hi @george-gca,

When a page is not found and the 404 template is triggered, the entire site localization doesn't take effect. For example, if you visit https://george-gca.github.io/multi-language-al-folio/pt-br/blog/category/blockquotes, the 404 page will be displayed in English, even if the language should not be English.
![Example](https://github.com/george-gca/multi-language-al-folio/assets/34124149/9f4a20f6-f19b-4020-92e1-ad6364561884)

**Comment by george-gca** [comment, marked as answer]
Unfortunately there is nothing I can do. The 404 page is automatically handled by the server, and apparently there can only be one of it (if you check the `pt-br` pages there is one [404.md](https://github.com/george-gca/multi-language-al-folio/blob/main/_pages/pt-br/404.md) page there, but this never gets called). Since the server handles this, it is impossible to check what was the language of the page that caused the 404, therefore it is an edge case that can't be handled correctly.

**Reply by DmitryRyumin** [reply]
Thanks for the answer!
So far I found a way out through JS to redirect to the correct 404 and then both 404s are involved.

```javascript
// Wait for the DOM content to be fully loaded
document.addEventListener('DOMContentLoaded', function() {
  // Get the current path
  var path = window.location.pathname;
  // Extract the language from the path
  var lang = path.split('/')[1];
  var not_found = path.split('/')[2];

  // Check if the page title is 'Page not found'
  var pageTitleElement = document.querySelector('.post-title');
  
  if (pageTitleElement && pageTitleElement.textContent.trim() === 'Page not found') {
    // Page title matches 'Page not found'
    if (lang === 'es') {
      // Redirect to /es/404
      window.location.href = '/es/404';
    } else {
      if (not_found !== '404') {
        // Redirect to /en/404
        window.location.href = '/en/404';
      }
    }
  }
});
```
